corehq/apps/accounting/async_handlers.py

```diff
@@ -212,9 +213,29 @@ def account_response(self):
     @property
     def plan_version_response(self):
         edition = self.data.get('additionalData[edition]')
-        plan_versions = SoftwarePlanVersion.objects.filter(
-            plan__edition=edition
-        )
+        visibility = self.data.get('additionalData[visibility]')
+        most_recent_version = self.data.get('additionalData[most_recent_version]')
+        if most_recent_version == 'True':
+            latest_versions_dates = SoftwarePlanVersion.objects.filter(
+                plan=OuterRef('pk'),
+                plan__edition=edition,
+                plan__visibility=visibility
+            ).order_by('-date_created').values('date_created')[:1]
+            latest_versions = SoftwarePlan.objects.filter(
+                edition=edition,
+                visibility=visibility
+            ).annotate(
+                latest_version_date=Subquery(latest_versions_dates)
+            ).values('id', 'name', 'latest_version_date')
+            plan_versions = SoftwarePlanVersion.objects.filter(
+                plan__in=[item['id'] for item in latest_versions],
+                date_created__in=[item['latest_version_date'] for item in latest_versions]
+            )
```

Comment:
I don't know `SoftwarePlanVersion`, but is it possible to have multiple entries for the same plan `id`? Looking at the model, I'd assume that there should only ever be one `is_active` entry per `id`. If so, perhaps we can query by checking for `is_active=True` rather than `date_created__in`?

Also, I'm nervous that neither of these fields (`date_created` or `is_active`) are indexed, which generally suggests we don't intend to do production queries on them. The production table only has ~1300 entries, so its probably not a big deal, but I still wonder if there was another intended method.

Reply:
> is it possible to have multiple entries for the same plan id

Yes. A software plan can have multiple versions

> If so, perhaps we can query by checking for is_active=True rather than date_created__in?

We can have multiple software plan versions per plan that is active. 

> if there was another intended method

I didn't find any method that I can leverage.